File: 0912-sort-an-array/0912-sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        if not nums:
            return nums
        min_val,max_val=min(nums),max(nums)
        counts=[0]*(max_val-min_val+1)
        for num in nums:
            counts[num-min_val]+=1
        j=0
        for i in range(len(counts)):
            while counts[i]>0:
                nums[j]=i+min_val
                j+=1
                counts[i]-=1
        return nums
        

File: 0912-sort-an-array/0912-sort-an-array.cpp
class Solution {
public:
    vector<int> sortArray(vector<int>& nums) {
        if(nums.empty()){
            return nums;
        }
        int min_val=*min_element(nums.begin(),nums.end());
        int max_val=*max_element(nums.begin(),nums.end());
        vector<int> counts(max_val-min_val+1);
        for(int num:nums){
            counts[num-min_val]++;
        }
        int j=0;
        for(int i=0;i<counts.size();i++){
            while(counts[i]-->0){
                nums[j++]=i+min_val;
            }
        }
        return nums;
    }
};